### PR TITLE
Pr Reminder tool

### PR DESCRIPTION
**Pull request by smg247**
See `README.md` for a description of this new tool.
This is quite a large PR, but I don't see anyway to break this up into a smaller, useful chunk.
The periodic to run this will be added in a forthcoming PR.

For [DPTP-2745](https://issues.redhat.com/browse/DPTP-2745)
/cc @openshift/test-platform

Screen Shot 2022-05-10 at 10 02 30 AM

### CODE REVIEW

File: cmd/pr-reminder/main.go [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+
+const (
+	recent = ":large_green_circle:"
+	normal = ":large_orange_circle:"
+	old    = ":red_circle:"
+)
+
+func (p prRequest) recency() string {
+	now := time.Now()
+	if p.Created.After(now.Add(-time.Hour * 24 * 2)) {
```
Comment (jmguzik) [inline cmd/pr-reminder/main.go:129]: maybe const out of this value?

File: cmd/pr-reminder/main.go [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+		return recent
+	} else if p.Created.After(now.Add(-time.Hour * 24 * 7)) {
```
Comment (jmguzik) [inline cmd/pr-reminder/main.go:131]: as above

File: cmd/pr-reminder/main.go [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+		}
+	}
+	if len(usersMissingGithubId) > 0 {
+		return nil, fmt.Errorf("no githubId found for user(s): %v", usersMissingGithubId)
+	}
+
+	return users, nil
+}
+
+func messageUser(user user, slackClient slackClient) error {
```
Comment (jmguzik) [inline cmd/pr-reminder/main.go:278]: This is just a suggestion: I would split this function into smaller chunks. It seems to be quite long.

Comment (smg247) [reply cmd/pr-reminder/main.go:278]: It is quite long, but its 90% simple boiler plate that the slack client requires to send a message. I think breaking it up would just make it even more time consuming to read.

**Review by jmguzik — APPROVED** [review]
Really nice work!

**Pull request merged**